internal/venice-test-common/src/main/java/com/linkedin/venice/utils/TestChunkingUtils.java

```diff
@@ -0,0 +1,183 @@
+package com.linkedin.venice.utils;
+
+import static com.linkedin.venice.kafka.protocol.enums.MessageType.*;
+
+import com.linkedin.venice.kafka.protocol.Delete;
+import com.linkedin.venice.kafka.protocol.GUID;
+import com.linkedin.venice.kafka.protocol.KafkaMessageEnvelope;
+import com.linkedin.venice.kafka.protocol.ProducerMetadata;
+import com.linkedin.venice.kafka.protocol.Put;
+import com.linkedin.venice.kafka.protocol.Update;
+import com.linkedin.venice.kafka.protocol.enums.MessageType;
+import com.linkedin.venice.message.KafkaKey;
+import com.linkedin.venice.pubsub.ImmutablePubSubMessage;
+import com.linkedin.venice.pubsub.api.PubSubMessage;
+import com.linkedin.venice.pubsub.api.PubSubTopicPartition;
+import com.linkedin.venice.serialization.KeyWithChunkingSuffixSerializer;
+import com.linkedin.venice.serialization.avro.AvroProtocolDefinition;
+import com.linkedin.venice.serialization.avro.ChunkedValueManifestSerializer;
+import com.linkedin.venice.storage.protocol.ChunkId;
+import com.linkedin.venice.storage.protocol.ChunkedKeySuffix;
+import com.linkedin.venice.storage.protocol.ChunkedValueManifest;
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+
+
+public class TestChunkingUtils {
+  private TestChunkingUtils() {
+    // Util class
```

Comment:
It got me confused for a minute if this is a test class vs utility class.
Can we replace `TestChunkingUtils` with `ChunkingUtils`? Looks like we add `Test` prefix to class names to denote test classes for those classes and this seems to conflict with the norm.

Additionally, if this is a util class, mark the class final for best practice.

Reply:
I also found it confusing. There is an existing `ChunkingUtils` class that is actually used in the production code, and I didn't want there to be two files with the same name. This file is within `venice-test-common` which indicates that it's for testing purposes, but I've found that it's simpler to infer from just the file name when searching file names using IntelliJ IDEA.

How about renaming it to `ChunkingTestUtils`? I've marked the class as `final` in 53008a1.